lib/helpers/stake/gateway_composer/Facilitator.js

```diff
@@ -31,20 +11,29 @@ class Facilitator {
   /**
    * Facilitator constructor object.
    *
-   * @param originWeb3 Origin chain web3 address.
-   * @param valueToken Value token contract address.
-   * @param brandedToken Branded Token contract address.
-   * @param gatewayComposer Gateway composer contract address.
-   * @param facilitator Facilitator address.
+   * @param {Web3} originWeb3 Origin chain web3 address.
+   * @param {string} valueToken Value token contract address.
+   * @param {string} brandedToken Branded Token contract address.
+   * @param {string} gatewayComposer Gateway composer contract address.
+   * @param {string} facilitator Facilitator address.
+   * @param {BtAbiBinProvider} abiBinProvider A class that provides ABIs and BINs.
    */
-  constructor(originWeb3, valueToken, brandedToken, gatewayComposer, facilitator) {
-    const oThis = this;
+  constructor(
+    originWeb3,
+    valueToken,
+    brandedToken,
+    gatewayComposer,
+    facilitator,
+    abiBinProvider,
```

Comment:
abiBinProvider doesn't need to be an argument in the constructor. It can be required and object can be created at top of file.
```
const AbiBinProvider = require('AbiBinProvider');
```
const abiBinProvider = new AbiBinProvider()

Reply:
It is bad practice to hard-code dependencies. Instead, they should be injected (called dependency injection). It makes the code extendable, easier to maintain, and easier to test (as you can pass mocks, etc. as dependencies).